packages/runner/lib/monitor.ts

```diff
@@ -0,0 +1,156 @@
+import os from 'os';
+import fs from 'fs';
+import type { ActivityLogMessage, NangoProps } from '@nangohq/shared';
+import * as superjson from 'superjson';
+import { fetch } from 'undici';
+import { getLogger } from '@nangohq/utils/dist/logger.js';
+
+const MEMORY_WARNING_PERCENTAGE_THRESHOLD = 75;
+const logger = getLogger('Runner');
+
+export class RunnerMonitor {
+    private runnerId: string;
+    private tracked: Map<number, NangoProps> = new Map<number, NangoProps>();
+    private baseUrl: string = '';
+    private idleMaxDurationMs = parseInt(process.env['IDLE_MAX_DURATION_MS'] || '') || 0;
+    private lastIdleTrackingDate = Date.now();
+    private lastMemoryReportDate: Date | null = null;
+    private idleInterval: NodeJS.Timeout | null = null;
```

Comment:
a bit out of scope but I renamed this logic RunnerMonitor and moved the idle logic into it. It is better here imho than floating in `runner/server.ts`